### ISSUE TITLE
expose `deps` in `buildImage`

### ISSUE THREAD

**Issue by blaggacao** [nlewo/nix2container#15]
Assume a container that needs healthchecks packaged.

- With `contents ? []` you are forced to invoke in a non-fully qualified manner:
  `/bin/cardano-wallet-network-sync-check`

- With `deps ? []` you would be able to invoke it in a fully qualified manner:
  `${healthChecks.wallet-network-sync}/bin/cardano-wallet-network-sync-check`

**Comment by blaggacao** [comment]
Maybe the best of both worlds would be if `contents` would symlink / or hardlink the nix store paths, so that an executable can be addressed in both ways?

**Comment by blaggacao** [comment]
More context: https://github.com/input-output-hk/bitte-cells/commit/5197fa8d8ee0b4b3b41936dcab0a484dfa2fbe29

**Comment by nlewo** [comment]
@blaggacao I would prefer to keep the current `contents` implementation because i think symlinking storepath is quite fragile. To avoid symlink issues, in `dockerTools.buildImage`, the `contents` closure exists in the `/nix/store` and the toplevel dependency of each `contents` derivation is copied in `/` :/

So, i would prefer to assume the toplevel dependency of each `contents` derivation is just moved to `/`. For instance, this works pretty well to create [`/var`](https://github.com/nlewo/nix2container/blob/fc3f6eb027755ba6954b73156c04adc7502e542c/examples/nginx.nix#L33). But, it is still possible to use [`symlinkJoin`](https://github.com/nlewo/nix2container/blob/fc3f6eb027755ba6954b73156c04adc7502e542c/examples/bash.nix#L9) to expose stuffs in `/bin`.

Regarding your use case, another approach would be to add an environment variable to your image config. Something such as:
```
nix2container.buildImage.config.env = [ "HEALTHCHECK=${healthcheck...}" ]
```
This way, the `healthcheck` storepath is in the Nix store and you could use the fully qualified invocation.
Could that work for you?

Ultimately, we could expose `buildImage.deps`, but I would prefer to keep the API minimal.

**Comment by blaggacao** [comment]
> Ultimately, we could expose buildImage.deps, but I would prefer to keep the API minimal.

Would that still be an option? I think this would be the cleanest solution in this case?

It is decently reasonable that deps in a nix context refer to store paths.

**Comment by nlewo** [comment]
Yep, we could add this argument if it is the best way to address your use-case.
